theano/tensor/nlinalg.py

```diff
@@ -625,9 +625,12 @@ def perform(self, node, inputs, outputs):
         (x,) = inputs
         (w, u, v) = outputs
         assert x.ndim == 2, "The input of svd function should be a matrix."
-        w[0], u[0], v[0] = self._numop(x,
-                                       self.full_matrices,
-                                       self.compute_uv)
+        if self.compute_uv:
+            w[0], u[0], v[0] = self._numop(x,
+                                           self.full_matrices,
+                                           self.compute_uv)
+        else:
+            u[0] = self._numop(x, self.full_matrices, self.compute_uv)
```

Comment:
This change show that we need to test compute_uv. It is probably not tested, as the old code should crash.

Also, if we compute only 1 output, the make_node should have only 1 output.